### PR TITLE
Trim trailing whitespace and ensure trailing commas using pre-commit

### PR DESCRIPTION
**Pull request by sloria**
## Purpose

Enforce consistent styling and prevent flake8 errors from trailing whitespace.

We've been informally enforcing the "trailing comma" rule. Why not have computers do the work for us?

## Changes

Using pre-commit:

* Automatically trim whitespace
* Automatically enforce consistency in formatting of lists in Python (trailing comma).

## QA Notes

No QA

## Documentation

None

## Side Effects

Possible merge conflicts with other PRs due to the size of this diff.

## Ticket

No ticket

### CODE REVIEW

**Comment by sloria** [comment]
The failing "push" build is invalid because it's checking the wrong branch (CenterForOpenScience/feature/more-pre-commit vs. sloria/feature/more-pre-commit). Travis is passing on my fork.

File: addons/base/tests/models.py [hunk 1]
```diff
@@ -6,9 +6,11 @@
 from django.utils import timezone
 from framework.auth import Auth
 from framework.exceptions import HTTPError
-from nose.tools import (assert_equal, assert_false, assert_in, assert_is,
-                        assert_is_none, assert_not_in, assert_raises,
-                        assert_true)
+from nose.tools import (
+    assert_equal, assert_false, assert_in, assert_is,
+    assert_is_none, assert_not_in, assert_raises,
+    assert_true
```
Comment (mfraezz) [inline addons/base/tests/models.py]: Trailing commas not enforced on imports?

**Comment by sloria** [comment]
Good point. I opened an issue: https://github.com/asottile/add-trailing-comma/issues/52

File: addons/base/tests/models.py (hunk 1 above)
Comment (sloria) [reply addons/base/tests/models.py]: This is now supported. I upgraded the hook version in my latest commit.

**Comment by sloria** [comment]
This is updated with the latest develop.

Also, to reduce the pain of conflicts (for both me and others with open PRs), I constrained the `trailing-comma` hook to only format the `api/` directory for now.

**Pull request merged**